src/app/services/backend.service.ts

```diff
@@ -92,13 +96,26 @@ export class BackendService{
 
 constructor(
   private http: HttpClient,
+  private auth: AuthService,
 ){}
 
 customerLogin(username, password){
-  const customerUrl = this.baseUrl + "/customers";
-  return this.http.post(customerUrl, { username: username, password:password}).toPromise();
+  const customerUrl = this.baseUrl + "api/customers";
+  return this.http.post(customerUrl, { username: username, password:password}).toPromise()
+.then((customer)=>{
```

Comment:
This part here, the `.then` down to the end of the `customerLogin` function should really be handled in the auth service. Same for the vendor side. You won't be getting the password back from the backend, and even if you did, it would be the hashed password thus it will never match what the user typed in. Just getting a 200 back from the server is enough to say that the password matched.